### ISSUE TITLE
Release manager should review lucene benchmarks before building release candidates

### ISSUE THREAD

**Issue by iverase** [apache/lucene#11827]
### Description

We were about to release Lucene 9.4.0 with an important performance regression. This regression was showing up in the benchmarks but obviously we missed reviewing them. The regression was actually caught during the daily monitoring of Elasticsearch development instead.

Speaking to other Lucene committers we thought that it may be a good idea to add a new task to the release manager handbook to review the Lucene benchmarks before building the release candidates. Obviously we are not expecting the release manager to find all regressions but something as obvious as the one in Lucene 9.4.0 would have been possible.

wdyt?

**Comment by rmuir** [comment]
i think a good step is to just make sure that we review the benchmark results before releasing and send an email out about anything that looks suspicious.

**Comment by msokolov** [comment]
yup. Possibly too if Mike M is bored he could implement an alarming system :) or export the data somehow so we could bolt one on the side?

**Comment by uschindler** [comment]
I fully agree, some checks should be done.
But here are a few bits that came into my mind:
- The @mikemccand benchmarks are running against main branch only. So the first check should be *before* creating the release branch, so we know that the start of branch is "clean".
- Anything else that gets included into release branch should be monitored by the RM if it negatively affects perf on main.
- Is there a benchmark available for the 9.x branch? If it would be configurable / modular we could have a base branch (against main and in parallel against 9.x). When a release branch is created we could compare the changes on the release branch with the baselines!?

These are just ideas, but basically the important thing would be: Before creating the release branch do a quick check through the benchmark suite on Mike's web page and find potential issues. During the release process when stuff is merged into release branch monitor all merged changes manually on main's benchmark.

**Comment by mikemccand** [comment]
> yup. Possibly too if Mike M is bored he could implement an alarming system :) or export the data somehow so we could bolt one on the side?

Actually I rather like the alarm idea.  Maybe we could start with something simple -- more than X% change over Y data points?  Maybe for starters the nightly bench just writes these alarms in (blinking red font, heh) at the [top of the index page](https://home.apache.org/~mikemccand/lucenebench/)?  If it seems like that is not too noisy, maybe we find some stronger way to "notify" us.

**Comment by mikemccand** [comment]
This was a spinoff from #11824.

**Comment by navneet1v** [comment]
@iverase Where are we seeing the degradation? We are seeing the degradation in K-NN Queries between 9.3 and 9.4 release. I am trying to figure out if there are other clients of lucene who are also facing the same issue.

**Comment by iverase** [comment]
@navneet1v this issue was triggered by a regression on a geo query and was addressed before the release. I have no knowledge of any regression in vector search.
